crypto/evp/evp_lib.c

```diff
@@ -1058,6 +1063,33 @@ EVP_MD *EVP_MD_CTX_get1_md(EVP_MD_CTX *ctx)
     return md;
 }
 
+int EVP_MD_CTX_get_size_ex(const EVP_MD_CTX *ctx)
+{
+    const EVP_MD *md = EVP_MD_CTX_get0_md(ctx);
+    EVP_MD_CTX *c = (EVP_MD_CTX *)ctx;
+
+    if (EVP_MD_xof(md)
```

Comment:
This condition `EVP_MOD_xof(md)` is actually superfluous and incorrect - because for BLAKE2 (which is not XOF) we provide the settable and gettable OSSL_DIGEST_PARAM_SIZE parameter.

So if there are gettable params that include the OSSL_DIGEST_PARAM_SIZE parameter we should call get_params() regardless of XOF. Otherwise we can fallback to the old EVP_MD_get_size() call.

Then there is the issue with the old FIPS provider compatibility because there SHAKE will have no gettable_ctx_param but it will return wrong size. I assume we have to live with that.

Reply:
This was not resolved!

Reply:
Updated